apps/ui/src/Communities/CommunityCard/StatusButton.tsx

```diff
@@ -17,12 +17,15 @@ const StatusButton: React.FC<CommunityCardProps> = ({
   setJoined,
   status,
 }) => {
+
+  const isBlocked = status === UserStatus.REJECTED;
+
   const renderIcon = useCallback(() => {
```

Comment:
create a Record<UserStatus, Component> and just return record[status], then you just need to check that status exists, reducing all of the if/else statements